CMakeLists.txt

```diff
@@ -339,8 +342,21 @@ if(EMSCRIPTEN)
     add_compile_flag("-sDISABLE_EXCEPTION_CATCHING=0")
     add_link_flag("-sDISABLE_EXCEPTION_CATCHING=0")
   endif()
-  # make the tools immediately usable on Node.js
-  add_link_flag("-sNODERAWFS")
+  # In the browser, there is no natural place to provide commandline arguments
+  # for a commandline tool, so let the user run the main entry point themselves
+  # and pass in the arguments there.
+  if (BUILD_FOR_BROWSER)
+    add_link_flag("-sENVIRONMENT=web,worker")
+    add_link_flag("-sINVOKE_RUN=0")
+    add_link_flag("-sEXPORTED_RUNTIME_METHODS=run,callMain,FS")    
+    add_link_flag("-sMODULARIZE")
+    add_link_flag("-sEXPORT_ES6")
+    add_link_flag("-sFILESYSTEM")
+    add_link_flag("-sFORCE_FILESYSTEM")
+    # On Node.js, make the tools immediately usable.
+    else()
+    add_link_flag("-sNODERAWFS")
+  endif()
```

Comment:
Indentation looks wrong on these last three lines?

Reply:
Ooops, fixed in https://github.com/WebAssembly/binaryen/pull/6131/commits/049abfcbd1c77a6f24e8f99d94bf37c0b6d0c8db.